packages/taquito/src/contract/rpc-contract-provider.ts

```diff
@@ -764,6 +765,75 @@ export class RpcContractProvider extends Provider implements ContractProvider, S
 
     return batch;
   }
+
+  /**
+   *
+   * @description Stake founds. Will sign and inject an operation using the current context
+   *
+   * @returns An operation handle with the result from the rpc node
+   *
+   * @param stake operation parameter
+   */
+  async stake(params: StakingParams) {
+    const estimate = await this.estimate(params, this.estimator.stake.bind(this.estimator));
+    const source = await this.signer.publicKeyHash();
```

Comment:
with previous comment that `StakingParmas` having optional source this line can be replaced by `const publicKeyHash = await this.signer.publicKeyHash();
const source = params.source ?? publicKeyHash;`